Fix potential race condition with suspend.

oliverchang opened the pull request:
The previous approach led to frequent race conditions, where the first power button press
to resume my device causes the device to instantly go to sleep again. This seems to be
because of some timing issues with the fallback `systemctl suspend` call that happens after
the initial steam `-ifrunning steam://shortpowerpress` call. 

Instead, we check explicitly if we're running in deckui mode, and don't do the
`systemctl suspend` call if so.

Tested that this works as expected in:
- Deck UI sessions
- Desktop sessions (with and without Steam running)

pastaq reviewed (changes requested):
Looks good, thanks for submitting this. Can you add some brief code comments to explain how we're getting the steam_cmd and how it's used? After that it should be gtg.

oliverchang commented:
Thanks for reviewing @pastaq ! I've added some more comments, and also changed to a cleaner alternative for getting the cmdline (by just reading /proc instead of calling out to `ps`).

pastaq commented:
Awesome. I'll do some final tests tonight and merge if all goes well. Thanks!
